nailgun/entity_mixins.py

```diff
@@ -717,3 +749,99 @@ def create(self, create_missing=None):
+        Check the server's response for errors and decode the response.
+    :meth:`update`
+        Create a :class:`nailgun.entity_mixins.Entity` object representing
+        the created entity and populate its fields.
+
+    See the individual methods for more detailed information.
+
+    """
+
+    def update_payload(self, fields=None):
+        """Create a payload of values that can be sent to the server.
+
+        By default, this method behaves just like :func:`_payload`. However,
+        one can also specify a certain set of fields that should be returned.
+        For more information, see :meth:`update`.
+
+        """
+        values = self.get_values()
+        if fields is not None:
+            values = {field: values[field] for field in fields}
+        return _payload(self.get_fields(), values)
+
+    def update_raw(self, fields=None):
+        """Update the current entity.
+
+        Make an HTTP PUT call to ``self.path('base')``. The request payload
+        consists of whatever is returned by :meth:`update_payload`. Return the
+        response.
+
+        :param fields: See :meth:`update`.
+        :return: A ``requests.response`` object.
+
+        """
+        return client.put(
+            self.path('self'),
+            self.update_payload(fields),
+            **self._server_config.get_client_kwargs()
+        )
+
+    def update_json(self, fields=None):
+        """Update the current entity.
+
+        Call :meth:`update_raw`. Check the response status code, decode JSON
+        and return the decoded JSON as a dict.
+
+        :param fields: See :meth:`update`.
+        :return: A dict consisting of the decoded JSON in the server's
+            response.
+        :raises: ``requests.exceptions.HTTPError`` if the response has an HTTP
+            4XX or 5XX status code.
+        :raises: ``ValueError`` If the response JSON can not be decoded.
+
+        """
+        response = self.update_raw(fields)
+        response.raise_for_status()
+        return response.json()
+
+    def update(self, fields=None):
+        """Update the current entity.
+
+        Call :meth:`update_json`, use the response to populate a new object
+        of type ``type(self)`` and return that object.
+
+        This method requires that
+        :meth:`nailgun.entity_mixins.EntityReadMixin.read` or some other
+        identical method be available on the current object. A more thorough
+        explanation is available at
+        :meth:`nailgun.entity_mixins.EntityCreateMixin.create`.
+
+        :param fields: An iterable of field names. Only the fields named in
+            this iterable will be updated. No fields are updated if an empty
+            iterable is passed in. All fields are updated if ``None`` is passed
+            in.
+        :raises: ``KeyError`` if asked to update a field but no value is
+            available for that field on the current entity.
+
+        """
+        return self.read(attrs=self.update_json(fields))
```

Comment:
We must ensure that all entities that uses UpdateMixin have the ReadMixin :smile: